File: geophone-python/REST/gp_response.py

from dataclasses import dataclass
from typing import Optional, List

@dataclass
class Provider:
    """Represents provider information returned by the GP API."""
    Name: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    LineType: Optional[str] = None
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None
    Quality: Optional[str] = None

    def __str__(self) -> str:
        return (f"ProviderOutput: Name = {self.Name}, City = {self.City}, State = {self.State}, "
                f"LineType = {self.LineType}, Latitude = {self.Latitude}, Longitude = {self.Longitude}, "
                f"Quality = {self.Quality}")

@dataclass
class Contact:
    """Represents contact information returned by the GP API."""
    Name: Optional[str] = None
    Address: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    Zip: Optional[str] = None
    Type: Optional[str] = None

    def __str__(self) -> str:
        return (f"ContactOutput: Name = {self.Name}, Address = {self.Address}, City = {self.City}, "
                f"State = {self.State}, Zip = {self.Zip}, Type = {self.Type}")

@dataclass
class Error:
    """Represents error information returned by the GP API."""
    Desc: Optional[str] = None
    Number: Optional[str] = None
    Location: Optional[str] = None

    def __str__(self) -> str:
        return f"Desc: {self.Desc} Number: {self.Number} Location: {self.Location}"

@dataclass
class GPResponse:
    """Represents the full response from the GP API."""
    providers: Optional[List[Provider]] = None
    contacts: Optional[List[Contact]] = None
    error: Optional[Error] = None

    def __str__(self) -> str:
        providers_str = ", ".join(str(p) for p in self.providers) if self.providers else "null"
        contacts_str = ", ".join(str(c) for c in self.contacts) if self.contacts else "null"
        error_str = str(self.error) if self.error else "null"
        return f"GP GPResponse: Providers = {providers_str}, Contacts = {contacts_str}, Error = {error_str}"

File: geophone-nodejs/src/SOAP/gp-response.js
export class GPResponse {
  constructor(data) {
    const providersArray = data.Providers?.Provider || [];
    const contactsArray = data.Contacts?.Contact || [];

    this.Providers = Array.isArray(providersArray)
      ? providersArray.map(p => new Provider(p))
      : [new Provider(providersArray)];

    this.Contacts = Array.isArray(contactsArray)
      ? contactsArray.map(c => new Contact(c))
      : [new Contact(contactsArray)];

    this.Error = data.Error ? new Error(data.Error) : null;
  }

  toString() {
    const providers = this.Providers.length ? this.Providers.map(p => p.toString()).join(', ') : 'null';
    const contacts = this.Contacts.length ? this.Contacts.map(c => c.toString()).join(', ') : 'null';
    const error = this.Error ? this.Error.toString() : 'null';
    return `GPResponse: Providers = ${providers}, Contacts = ${contacts}, Error = ${error}`;
  }
}
export class Provider {
  constructor(data) {
    this.Name = data.Name || '';
    this.City = data.City || '';
    this.State = data.State || '';
    this.LineType = data.LineType || '';
    this.Latitude = data.Latitude || '';
    this.Longitude = data.Longitude || '';
    this.Quality = data.Quality || '';
  }

  toString() {
    return `ProviderOutput: Name = ${this.Name}, City = ${this.City}, State = ${this.State}, LineType = ${this.LineType}, Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, Quality = ${this.Quality}`;
  }
}
export class Contact {
  constructor(data) {
    this.Name = data.Name || '';
    this.Address = data.Address || '';
    this.City = data.City || '';
    this.State = data.State || '';
    this.Zip = data.Zip || '';
    this.Type = data.Type || '';
  }

  toString() {
    return `ContactOutput: Name = ${this.Name}, Address = ${this.Address}, City = ${this.City}, State = ${this.State}, Zip = ${this.Zip}, Type = ${this.Type}`;
  }
}